### PR TITLE
docs: Add documentation for how to add a gitPrivateKey

### PR DESCRIPTION
**Pull request by nachtien**
Maybe because I'm not very familiar with GPG signing, but I really struggled with this step when I configured it on a gitlab runner.  I think the main issue I ran into was using the correct real name and email address, and possibly replacing all new lines with `\n`.  

I hope this makes it more clear for the next person.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/self-hosted-configuration.md
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:427]:
```suggestion
Next, we want to save the private key into an environment variable.
The following command will copy the private key to the clipboard while replacing all new lines with `\n`.
```

File: docs/usage/self-hosted-configuration.md
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:421]:
```suggestion
To generate a GPG key, run the following commands in the terminal:
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: docs/usage/self-hosted-configuration.md
Comment (rarkins) [inline docs/usage/self-hosted-configuration.md:425]: I'm not sure this is actually correct, is it? It should match the git committer, not the "official" bot email address? Also the public key needs to be uploaded somewhere publicly such as GitHub so that it can be verified.

**Review by rarkins — CHANGES_REQUESTED** [review]
I'm not sure this is formatting as intended? Here's the preview:

image

File: docs/usage/self-hosted-configuration.md
Comment (HonkingGoose) [reply docs/usage/self-hosted-configuration.md:425]: The `privateKey` docs for self-hosted users [^docs] also mentions GPG to create a private key.

The `privateKey` docs and this PR have some differences:

- The `privateKey` docs uses different commands:
  - Our docs: `gpg --full-generate-key`, this PR: `gpg --gen-key`
  - Our docs:  `gpg --armor --export-secret-keys YOUR_NEW_KEY_ID > renovate-private-key.asc` , this PR: `gpg --export-secret-keys --armor [your public key] | awk -v ORS='\\n' '1' | pbcopy`
- Our docs use `RENOVATE_PRIVATE_KEY` as variable name, this PR: `RENOVATE_GIT_PRIVATE_KEY`

I'll let you review these differences, I don't know enough about signing to help with this. 😉 

[^docs]: https://docs.renovatebot.com/self-hosted-configuration/#privatekey

Comment (rarkins) [reply docs/usage/self-hosted-configuration.md:425]: Those are different options

Comment (nachtien) [reply docs/usage/self-hosted-configuration.md:425]: I wrote this for a self hosted on Gitlab. This configuration worked for me. I didn't know that there could be other scenarios. Can someone advise what it should say?

Comment (HonkingGoose) [reply docs/usage/self-hosted-configuration.md:425]: I don't know enough about `gpg` or `pgp` to help out here. 😄 I'll let one of the maintainers help you. 😉

Comment (nachtien) [reply docs/usage/self-hosted-configuration.md:425]: @rarkins, does this email address also have to match what's in the renovate's [config](https://gitlab.com/renovate-bot/renovate-runner/-/blob/main/.gitlab-ci.yml#L4)?  That's what I why I have it as `bot@renovateapp.com`

Comment (rarkins) [reply docs/usage/self-hosted-configuration.md:425]: I think it needs to match the git _committer_, which is not the same as `gitAuthor` in Renovate config.

Comment (nachtien) [reply docs/usage/self-hosted-configuration.md:425]: git committer and `gitAuthor` really seem like the same thing. When I commit something, I'm the author. 

For brevity and to avoid confusion during setup, maybe it would be beneficial to keep them the same?

Comment (rarkins) [reply docs/usage/self-hosted-configuration.md:425]: Your first sentence is incorrect, please google it

Comment (rarkins) [inline docs/usage/self-hosted-configuration.md:425]:
```suggestion
Email Address: `yourbot@youremail.com`
```

**Comment by rarkins** [comment]
Closing due to inactivity

**Pull request closed without merging**
